### PR TITLE
Linear approach function moved to core

### LINKED ISSUE #81: Provide certain behavior as separate classes/functions
Many default states for the devices have very similar code, because essentially they are "approach a target value linearly with a given rate of change". There should be exactly one place in the code where this is implemented.

My suggestion would be to create a new module in `core` for this. Maybe even in a sub-module called `behavior` or something along those lines.

### PR DESCRIPTION
**Pull request by MichaelWedel**
This fixes #81.

There is now an additional `core.approaches` module, which contains a function `linear`. Others could be added easily. I tried implementing this as a class first with rate as a parameter to `__init__`, but the problem is that the rate is not always constant (see Linkam).

### CODE REVIEW

**Comment by OwenArnold** [comment]
I'll review this. Probably between meetings today.

File: core/approaches.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    :return: The new variable value.
+    """
+    sign = (target > current) - (target < current)
+
+    if not sign:
+        return current
+
+    new_value = current + sign * rate * dt
+
+    if sign * new_value > sign * target:
```
Comment (OwenArnold) [inline core/approaches.py:49]: Ok. So the function will automatically prevent overshooting, if say your `dt` is very coarse

**Comment by MikeHart85** [comment]
I'd already started looking at it yesterday, just wanted to test it with IBEX to make sure the Linkam still behaves as expected.

Something is not quite right with it, I'm looking into it now.

**Review by OwenArnold — COMMENTED** [review]
I'll leave @MikeHart85  to do the rest of review. As expected code sharing is a lot better. [GPA 3.06](https://codeclimate.com/github/DMSC-Instrument-Data/plankton/compare/approach_function) with the changes

**Comment by MichaelWedel** [comment]
Hm, the unit tests still pass for the linkam and the chopper also seemed to work as expected. Let me know if you find out what it is.

**Comment by MikeHart85** [comment]
The issue wasn't related to this change... it was just a typo in the set_limit command for the Linkam ("L[number]" instead of "L1[number]").

Unit tests wouldn't have seen that since they don't issue the commands but call the interface directly... We'll need to set up system tests to catch things like that.

**Comment by MikeHart85** [comment]
All looks good to me now, should be ready to merge?

**Comment by OwenArnold** [comment]
@MikeHart85. Your changes look fine. I say merge it.

**Comment by MichaelWedel** [comment]
Ah okay, I must have missed that. Good thing you discovered it before 1.0 :)

**Pull request merged**